### PR TITLE
feat: Minimal reactive volume viewer

### PR DESCRIPTION
**Pull request by manzt**
Implements proposed API in niivue/ipyniivue#51 using the example from basic multi-planar as an example.

```py
from ipyniivue_experimental import AnyNiivue, SliceType

volumes = [
  { "path": "images/mni152.nii.gz", "colormap": "gray", "visible": True, "opacity": 1.0 },
  { "path": "images/hippo.nii.gz", "colormap": "red", "visible": True, "opacity": 1 },
]
nv = AnyNiivue(slice_type=SliceType.MULTIPLANAR)
nv.load_volumes(volumes)
nv
```

The volumes are reactive for `colormap` and `opacity`. Meaning the above will change interactively with:

```py
v.volumes[0].colormap = "blue"
nv.volumes[0].opacity = 0.2
```

Finally, I added a script in `script/generate_options_mixin.py` to generate the `_options_mixin.py` file. This takes the defaults for all the viewer options and exposes reactive getters/setters for the values.

Thus, you can change viewer-level properties like `slice_type` programatically:

```py
nv.slice_type = SliceType.SAGITTAL # updates view
```

### CODE REVIEW

File: src/ipyniivue_experimental/_options_mixin.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
+# This file is automatically generated by scripts/generate_options_mixin.py
+# Do not edit this file directly
```
Comment (manzt) [inline src/ipyniivue_experimental/_options_mixin.py:2]: this file is automatically generated with getters/setters for the front-end properties.

**Comment by manzt** [comment]
Screen Recording 2024-02-20 at 11 25 42 PM

**Comment by manzt** [comment]
The default values for the `OptionsMixin` come from niivue DEFAULT_OPTIONS. I think that several of these properties don't really make sense on the Python side. One thing that would be nice is to annotate those options on the JS comments with extra metadata if we should ignore, and then this whole process can be automated (deriving from the JS as the source of truth).

**Comment by kolibril13** [comment]
Just tested this notebook, and I'm impressed with what you implemented here!
image

Also, the auto-generated generate_options_mixin works nicely, I've tried a few options from there.

As this pr is quite large, I suggest that we merge this one soon, and then discuss further steps on discord.

Feel free to also delete the current notebooks and add the `Untitled1.ipynb` that you show in the demo.

**Comment by manzt** [comment]
I think we can add the demo later. I prefer not to check in notebooks right now as they cause many issues with version control. Ideally, we should have the examples from niivue/ipyniivue#50 organized nicely into some notebooks like (examples, etc).

**Comment by manzt** [comment]
@kolibril13 I don't have write access to this repo so I cannot merge.

**Comment by kolibril13** [comment]
Nice, Ive merged this pr now from the GitHub app. I will add you as a repo admin once I am back home

**Pull request merged**